Cavein254/savannah-interview
# Feature/server:Implement Google Authentication with Passport and MongoDB

## Description:

This PR introduces Google OAuth 2.0 authentication using the `passport-google-oauth20` strategy. It also integrates MongoDB to store authenticated user details.
## Key Features:
1.  Google OAuth Integration:
 * Configures Passport with Google OAuth2.0 strategy.
 * Retrieves user profile information (email, name, avatar).

2. MongoDB Integration:
 * Stores authenticated user details in a MongoDB database.

## How to test:
1. check into current branch
2. add `GOOGLE_CLIENT_ID, GOOGLE_CLIENT_SECRET, and GOOGLE_CALLBACK_URL` to your environment variables
3. do `npm install` and `npm run start` to run this PR locally
4. navigate to `http://localhost:4000/auth/google`
5. google auth screen should appear and redirect you to the dashboard

## Screenshots
![Screenshot 2024-11-30 103435](https://github.com/user-attachments/assets/916944b1-f573-4458-b6cd-f4bc2373f63d)


![Screenshot 2024-11-30 103457](https://github.com/user-attachments/assets/9eb1980a-0e39-4441-a832-4d59a3c2263d)
